manager/syscalls.go

```diff
@@ -11,29 +11,86 @@ import (
 	"github.com/DataDog/ebpf/internal"
 )
 
-func perfEventOpenTracepoint(id int, progFd int) (*internal.FD, error) {
+// perfEventOpenWithProbe - Kernel API with e12f03d ("perf/core: Implement the 'perf_kprobe' PMU") allows
+// creating [k,u]probe with perf_event_open, which makes it easier to clean up
+// the [k,u]probe. This function tries to create pfd with the perf_kprobe PMU.
+func perfEventOpenWithProbe(name string, offset, pid int, sectionPrefix string, referenceCounterOffset uint64) (int, error) {
+	var err error
+	attr := unix.PerfEventAttr{
+		Sample: 1,
+		Wakeup: 1,
```

Comment:
Are `Sample` and `Wakeup` needed? I've not used them to attach kprobes before.

Reply:
I'm following what BCC does, we also set those parameters to 1 in the "old" method (which also comes from BCC).

With the new ABI: https://github.com/iovisor/bcc/blob/887e05abd49c6587ede708de6b7040830d90e075/src/cc/libbpf.c#L875-L876

With the old one:
https://github.com/iovisor/bcc/blob/887e05abd49c6587ede708de6b7040830d90e075/src/cc/libbpf.c#L951-L952

Reply:
`libbpf` doesn't seem to use them: https://github.com/libbpf/libbpf/blob/99bc176337071326d7d6b9552142eb6a0450fc88/src/libbpf.c#L9735-L9738

Reply:
I don't have any particular preference, what do you think we should do ? Follow `libbcc` or `libbpf` ? :)

Reply:
I would consider `libbpf` a more authoritative source 😁

Reply:
All right, I delete them for the new ABI :)